src/udiskslinuxdevice.c

```diff
@@ -273,6 +275,15 @@ probe_ata (UDisksLinuxDevice  *device,
   UDisksAtaCommandInput input = {0};
   UDisksAtaCommandOutput output = {0};
 
+  if (!force_probe
+#ifndef HAVE_UDEV_257
```

Comment:
This might be worth rewriting (with `ifdef-else` maybe) a bit to make it more easy to understand when the probing is not necessary (because newer udev already has everything we need).

Reply:
Improved.